src/ai/backend/manager/repositories/model_serving/repository.py

```diff
@@ -650,8 +663,8 @@ async def update_auto_scaling_rule_validated(
                 for key, value in fields_to_update.items():
                     setattr(rule, key, value)
 
-                return rule
-            except NoResultFound:
+                return rule.to_data()
+            except ObjectNotFound:
                 return None
```

Comment:
This looks like a good direction to produce an error, let's clean it up in the next sprint.